src/digraph.rs

```diff
@@ -2265,6 +2266,150 @@ impl PyDiGraph {
         Ok(NodeMap { node_map: out_map })
     }
 
+    /// Substitute a set of nodes with a single node.
+    ///
+    /// :param set to_replace: A set of nodes to be removed and replaced
+    ///     by the specified node.
+    /// :param int node: An existing node to use as the replacement.
+    /// :param bool force_check_cycle: If set to ``True``, validates
+    ///     that the substitution will not introduce cycles before
+    ///     modifying the graph, even if cycle check is disabled for
+    ///     this instance of :class:`retworkx.PyDiGraph`. Otherwise,
+    ///     the value of member ``check_cycle`` is inherited.
+    /// :raises DAGWouldCycle: The cycle check is enabled and the
+    ///     substitution would introduce cycle(s).
+    /// :raises ValueError: ``to_replace`` is empty or ``node``
+    ///     has existing connections.
+    #[pyo3(
+        text_signature = "(self, to_replace, node, /, force_check_cycle=None)"
+    )]
+    fn substitute_nodes_with_node(
+        &mut self,
+        py: Python,
+        to_replace: HashSet<usize>,
```

Comment:
I'm wondering if we want this to be `Vec<usize>` instead. Python users tend to use lists (it also enables users to pass any [sequence type](https://pyo3.rs/v0.15.0/conversions/tables.html#argument-types) not just a list) more than sets and `Vec<T>` iterates faster than a set. Since the first thing we do with `to_replace` is loop over it check each index is valid and convert it to a `HashSet<NodeIndex>` I think making the argument a vec will work better here

Reply:
Okay by me!

Reply:
Done in c8699ae.